core/field.js

```diff
@@ -1254,11 +1257,17 @@ Field.prototype.DEFAULT_VALUE = null;
  */
 Field.NBSP = '\u00A0';
 
+/**
+ * A class used to provide a type for SKIP_SETUP.
+ */
+Field.Sentinel = class {};
```

Comment:
> Are you doing this just to avoid having to separately import it in subclass modules for type-checking purposes? I'm not convinced that is a good enough reason.

To be honest, did not have a reason haha.

> (Indeed: this class doesn't even need to be declared in this module; a general-purpose utility module could provide it.)

I'm a bit ambivalent about providing `Sentinel` in a utils module, because I don't really want to encourage our use of this pattern in other places. However, I agree that it is general-purpose and doesn't belong in Field, so I've moved it into a utility file anyway.